src/nvim/os/input.c

```diff
@@ -302,16 +302,16 @@ static InbufPollResult inbuf_poll(int ms)
   return eof ? kInputEof : kInputNone;
 }
 
-static void read_cb(RStream *rstream, void *data, bool at_eof)
+static void read_cb(RStream *rstream, RBuffer *buf, void *data, bool at_eof)
 {
   if (at_eof) {
     eof = true;
   }
 
-  char *buf = rbuffer_read_ptr(read_buffer);
-  size_t buf_size = rbuffer_pending(read_buffer);
-  (void)rbuffer_write(input_buffer, buf, buf_size);
-  rbuffer_consumed(read_buffer, buf_size);
+  RBUFFER_UNTIL_READ_OR_EMPTY(read_buffer, ptr, len) {
+    (void)rbuffer_write(input_buffer, ptr, len);
```

Comment:
This bug is already present(#1813) right? I intend to fix before #2371, which will be done by allocating memory for new input.